pkg/controllers/user/monitoring/operatorHandler.go

```diff
@@ -61,6 +68,74 @@ func (h *operatorHandler) sync(key string, obj *mgmtv3.Cluster) (runtime.Object,
 	return obj, nil
 }
 
+func (h *operatorHandler) syncProject(key string, project *mgmtv3.Project) (runtime.Object, error) {
+	if project == nil || project.DeletionTimestamp != nil || project.Spec.ClusterName != h.clusterName {
+		return project, nil
+	}
+
+	clusterID := project.Spec.ClusterName
+	cluster, err := h.cattleClusterClient.Get(clusterID, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to find Cluster %s", clusterID)
+	}
+
+	var newCluster *mgmtv3.Cluster
+	//should deploy
+	if project.Spec.EnableProjectMonitoring {
+		newObj, err := mgmtv3.ClusterConditionPrometheusOperatorDeployed.DoUntilTrue(cluster, func() (runtime.Object, error) {
+			cpy := cluster.DeepCopy()
+			return cpy, deploySystemMonitor(cpy, h.app)
+		})
+		if err != nil {
+			logrus.WithError(err).Info("sync cluster monitor operator error")
+		}
+		newCluster = newObj.(*mgmtv3.Cluster)
+	} else { // should withdraw
+		areAllOwnedProjectMonitoringDisabling, err := allOwnedProjectsMonitoringDisabling(h.cattleProjectClient)
+		if err != nil {
+			return nil, err
+		}
+		isAlertingDisabling := mgmtv3.MonitoringConditionAlertmaanagerDeployed.IsFalse(cluster) ||
+			mgmtv3.MonitoringConditionAlertmaanagerDeployed.GetStatus(cluster) == ""
+		isClusterMonitoringDisabling := mgmtv3.ClusterConditionMonitoringEnabled.IsFalse(cluster) ||
+			mgmtv3.ClusterConditionMonitoringEnabled.GetStatus(cluster) == ""
+		//status false and empty should withdraw. when status unknown, it means the deployment has error while deploying apps
+		isOperatorDeploying := !mgmtv3.ClusterConditionPrometheusOperatorDeployed.IsFalse(cluster)
+
+		if areAllOwnedProjectMonitoringDisabling && isAlertingDisabling && isClusterMonitoringDisabling && isOperatorDeploying {
+			newCluster = cluster.DeepCopy()
+			if err = withdrawSystemMonitor(h.app, cluster.Name); err != nil {
+				mgmtv3.ClusterConditionPrometheusOperatorDeployed.Unknown(newCluster)
+				mgmtv3.ClusterConditionPrometheusOperatorDeployed.ReasonAndMessageFromError(newCluster, err)
+			} else {
+				mgmtv3.ClusterConditionPrometheusOperatorDeployed.False(newCluster)
+			}
+		}
+	}
+
+	if newCluster != nil && !reflect.DeepEqual(newCluster, cluster) {
+		if _, err = h.cattleClusterClient.Update(newCluster); err != nil {
+			return nil, err
+		}
+	}
+
+	return project, nil
+}
+
+func allOwnedProjectsMonitoringDisabling(projectClient mgmtv3.ProjectInterface) (bool, error) {
+	ownedProjectList, err := projectClient.List(metav1.ListOptions{})
+	if err != nil {
+		return false, errors.Wrap(err, "failed to list all projects")
+	}
+
+	anyOwnedProjectsMonitoringEnabling := false
+	for _, ownedProject := range ownedProjectList.Items {
+		anyOwnedProjectsMonitoringEnabling = ownedProject.Spec.EnableProjectMonitoring || anyOwnedProjectsMonitoringEnabling
```

Comment:
you can return false if ownedProject.Spec.EnableProjectMonitoring is true, and return true in the end of the function.